### DISCUSSION TITLE
Help required to connect with Kepware using pymodbus

### DISCUSSION THREAD

**Discussion by coder-maddy** [pymodbus-dev/pymodbus#1920] [Help] [answered]
Hello Community,

I am trying to connect with Kepware modbus serial using pymodbus.
But I am getting below error.

`[Connection] Failed to connect[ModbusSerialClient(<pymodbus.framer.rtu_framer.ModbusRtuFramer object at 0x000002565A01A8D0> baud[9600])]`

Here is the code for connecting.

`with ModbusSerialClient('COM50', baudrate=9600) as client:
            client.send(meterReading.get_serial_data())`

Here is my kepware configuration

![image](https://github.com/pymodbus-dev/pymodbus/assets/16460793/308fcd6e-bb7f-44b9-8895-18fb007c449b)

Any help will be appreciated.
Thanks

**Comment by janiversen** [comment, marked as answer]
Are you sure "com50" is correct ? I know it writes 50 as Id, but that does not mean that it translates to COM50.

Secondly you only set baud rate, I think you need to set more serial parameters.

Thirdly client.send is not an external method, please look at the documentation.

Please use our debug call, to get more information (see examples in the documentation).

**Comment by janiversen** [comment]
Still a problem ?

**Reply by coder-maddy** [reply]
It’s sorted @janiversen 
Thanks
